util/file_reader_writer.cc

```diff
@@ -514,6 +514,31 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
     return Status::OK();
   }
 
+  virtual Status ReadaheadBackwards(uint64_t offset) {
```

Comment:
Why don't you let block based table reader to provide offset and length? And why it is called ReadaheadBackwards()? Can we just call Prefetch(offset, length?)?